src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerParameterSymbol.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class FunctionPointerParameterSymbol : ParameterSymbol
+    {
+        private readonly FunctionPointerMethodSymbol _containingSymbol;
+
+        public FunctionPointerParameterSymbol(TypeWithAnnotations typeWithAnnotations, RefKind refKind, int ordinal, FunctionPointerMethodSymbol containingSymbol, ImmutableArray<CustomModifier> refCustomModifiers, ImmutableArray<Location> locations)
+        {
+            TypeWithAnnotations = typeWithAnnotations;
+            RefKind = refKind;
+            Ordinal = ordinal;
+            _containingSymbol = containingSymbol;
+            RefCustomModifiers = refCustomModifiers;
+            Locations = locations;
+        }
+
+        public override TypeWithAnnotations TypeWithAnnotations { get; }
+        public override RefKind RefKind { get; }
+        public override int Ordinal { get; }
+        public override Symbol ContainingSymbol => _containingSymbol;
+        public override ImmutableArray<CustomModifier> RefCustomModifiers { get; }
+        public override ImmutableArray<Location> Locations { get; }
+        public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences => GetDeclaringSyntaxReferenceHelper<ParameterSyntax>(Locations);
+
+        public override bool Equals(Symbol other, TypeCompareKind compareKind)
+        {
+            if (!(other is FunctionPointerParameterSymbol param))
+            {
+                return false;
+            }
+
+            return Equals(param, compareKind, isValueTypeOverride: null);
+        }
+
+        internal bool Equals(FunctionPointerParameterSymbol other, TypeCompareKind compareKind, IReadOnlyDictionary<TypeParameterSymbol, bool>? isValueTypeOverride)
+            => EqualsContainingVerified(other, compareKind, isValueTypeOverride)
+               && _containingSymbol.Equals(other._containingSymbol, compareKind, isValueTypeOverride);
```

Comment:
> _containingSymbol.Equals(other._containingSymbol, compareKind, isValueTypeOverride); [](start = 17, length = 85)

Is this going to compare parameters? Is there a chance to go into a cycle? I think that it would be much simple if Equals.Override for the parameter symbol compared ordinal and containing FunctionPointerMethodSymbol. And the Equals implementation for FunctionPointerMethodSymbol compared parameters' properties other than ordinal, it would be fine to have a helper method for that in this class, just not called Equals.

Reply:
> Is there a chance to go into a cycle?

No, because the parent calls `EqualsContainingVerified`. That being said, I like the idea of the ordinal version, I'll change do that.

---
In reply to: [362685678](https://github.com/dotnet/roslyn/pull/40431#discussion_r362685678) [](ancestors = 362685678)